frontend/pkg/frontend/node_pool.go

```diff
@@ -105,13 +111,28 @@ func (f *Frontend) CreateOrUpdateNodePool(writer http.ResponseWriter, request *h
 		// the Tags map to remain nil so we can see if the request
 		// body included a new set of resource tags.
 
+		hcpNodePool.SystemData = resourceDoc.SystemData
+		hcpNodePool.Properties.ProvisioningState = resourceDoc.ProvisioningState
+
 		operationRequest = database.OperationRequestUpdate
 
 		// This is slightly repetitive for the sake of clarify on PUT vs PATCH.
 		switch request.Method {
 		case http.MethodPut:
+			// Initialize versionedRequestNodePool to include both
+			// non-zero default values and current read-only values.
+			reqNodePool := api.NewDefaultHCPOpenShiftClusterNodePool()
+
+			// Some optional create-only fields have dynamic default
+			// values that are determined downstream of this phase of
+			// request processing. To ensure idempotency, add these
+			// values to the target struct for the incoming request.
+			reqNodePool.Properties.Version.ID = hcpNodePool.Properties.Version.ID
```

Comment:
I'm worried that it's going to be hard to remember to add to this in future. Could we move the dynamic defaulfs into a new function like NewDefaultHCPOpenShiftClusterNodePoolWithDynamicDefaults that takes a HCPNodePoolDynamicDefaults just so it's more self-contained and obvious